tools/settings.py

```diff
@@ -71,6 +51,42 @@
 # mbed.org username
 MBED_ORG_USER = ""
 
+
+##############################################################################
+# User Settings (file)
+##############################################################################
+try:
+    # Allow to overwrite the default settings without the need to edit the
+    # settings file stored in the repository
+    from mbed_settings import *
+except ImportError:
+    pass
+
+
+##############################################################################
+# User Settings (env vars)
+##############################################################################
+_ENV_PATHS = ['ARM_PATH', 'GCC_ARM_PATH', 'GCC_CR_PATH', 'IAR_PATH']
```

Comment:
Perhaps we should prefix the `_ENV_PATHS` with `MBED`, so:

```
_ENV_PATHS = ['MBED_ARM_PATH', 'MBED_GCC_ARM_PATH', 'MBED_GCC_CR_PATH', 'MBED_IAR_PATH']
```

The reason is these variables are just general enough where they might conflict with some other environment variables on other people's machines.